<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <arg name="cloud_topic" default="/cloud" />
  <arg name="velocity_topic_in" default="/cmd_vel/autonomous" />
  <arg name="velocity_topic_out" default="/cmd_vel/capped" />
  <arg name="mode" default="$(env MOVEL_MODE)" doc="mode: [deployment, development]"/>

  <group if="$(eval mode == 'deployment')">    
    <node pkg="velocity_limiter" type="velocity_limiter_node" name="velocity_limiter" output="screen">
      <rosparam file="$(find velocity_limiter)/config/velocity_limiter.yaml"/>
      <rosparam file="$(env HOME)/.config/movel/config/velocity_limiter/config/velocity_limiter.yaml"/>
      <remap from="/cloud" to="$(arg cloud_topic)" />
      <remap from="/cmd_vel" to="$(arg velocity_topic_in)" />
      <remap from="/cmd_vel/capped" to="$(arg velocity_topic_out)" />
    </node>
  </group>

  <group unless="$(eval mode == 'deployment')">    
    <node pkg="velocity_limiter" type="velocity_limiter_node" name="velocity_limiter" output="screen">
      <rosparam file="$(find velocity_limiter)/config/velocity_limiter.yaml"/>
      <remap from="/cloud" to="$(arg cloud_topic)" />
      <remap from="/cmd_vel" to="$(arg velocity_topic_in)" />
      <remap from="/cmd_vel/capped" to="$(arg velocity_topic_out)" />
    </node>
  </group>

</launch>
